src/config-file.c

```diff
@@ -256,73 +256,91 @@ static GLogLevelFlags log_level_from_string(const gchar *log_level)
                        G_LOG_LEVEL_WARNING | G_LOG_LEVEL_MESSAGE;
 }
 
-Config* load_config_file(const gchar* config_file, GError** error)
+Config* load_config_file(const gchar *config_file, GError **error)
 {
-        Config *config = g_new0(Config, 1);
-
-        gint val_int;
+        g_autoptr(Config) config = NULL;
         g_autofree gchar *val = NULL;
-        g_autoptr(GKeyFile) ini_file = g_key_file_new();
+        g_autoptr(GKeyFile) ini_file = NULL;
         gboolean key_auth_token_exists = FALSE;
         gboolean key_gateway_token_exists = FALSE;
 
-        if (!g_key_file_load_from_file(ini_file, config_file, G_KEY_FILE_NONE, error)) {
+        g_return_val_if_fail(config_file, NULL);
+        g_return_val_if_fail(error == NULL || *error == NULL, NULL);
+
+        config = g_new0(Config, 1);
+        ini_file = g_key_file_new();
+
+        if (!g_key_file_load_from_file(ini_file, config_file, G_KEY_FILE_NONE,
+                                       error))
                 return NULL;
-        }
 
-        if (!get_key_string(ini_file, "client", "hawkbit_server", &config->hawkbit_server, NULL, error))
+        if (!get_key_string(ini_file, "client", "hawkbit_server",
+                            &config->hawkbit_server, NULL, error))
                 return NULL;
 
-        key_auth_token_exists = get_key_string(ini_file, "client", "auth_token", &config->auth_token, NULL, NULL);
-        key_gateway_token_exists = get_key_string(ini_file, "client", "gateway_token", &config->gateway_token, NULL, NULL);
+        key_auth_token_exists = get_key_string(ini_file, "client",
+                                               "auth_token",
+                                               &config->auth_token, NULL,
+                                               NULL);
+        key_gateway_token_exists = get_key_string(ini_file, "client",
+                                                  "gateway_token",
+                                                  &config->gateway_token, NULL,
+                                                  NULL);
         if (!key_auth_token_exists && !key_gateway_token_exists) {
-                g_set_error(error, 1, 4, "Neither auth_token nor gateway_token is set in the config.");
+                g_set_error(error, G_KEY_FILE_ERROR,
+                            G_KEY_FILE_ERROR_INVALID_VALUE,
+                            "Neither auth_token nor gateway_token is set in the config.");
+                return NULL;
+        }
+        if (key_auth_token_exists && key_gateway_token_exists) {
```

Comment:
This changes behavior, thus please briefly describe that and why this should be an error instead in the commit message.

Reply:
I've moved this change into a dedicated commit with an appropriate commit message.